commands/template_store_pull.go

```diff
@@ -0,0 +1,66 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultTemplateList = "https://raw.githubusercontent.com/martindekov/store/martindekov/unofficial_templates_json/templates.json"
+)
+
+func init() {
+	templateStoreCmd.PersistentFlags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing templates?")
+	templateStoreCmd.PersistentFlags().BoolVar(&pullDebug, "debug", false, "Enable debug output")
+
+	templateStoreCmd.AddCommand(templateStorePullCmd)
+}
+
+var templateStorePullCmd = &cobra.Command{
+	Use:   `pull [TEMPLATE_NAME]`,
+	Short: `Pull templates from store`,
+	Long:  `Pull templates from store supported by openfaas or openfaas-incubator organizations`,
+	Example: `  faas-cli template store pull ruby-http ...
+  faas-cli template store pull openfaas/go openfaas-incubator/node10-express ...`,
+	RunE: runTemplateStorePull,
+}
+
+func runTemplateStorePull(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("\nNeed to specify one of the store templates check aviable ones by running the command:\n\nfaas-cli template store list\n")
+	} else {
+		storeTemplates, templatesErr := getTemplateInfo(DefaultTemplateList)
+		if templatesErr != nil {
+			return fmt.Errorf("error while fetching templates from store: %s", templatesErr)
+		}
+
+		for _, template := range args {
+			found := false
+			for _, storeTemplate := range storeTemplates {
+				sourceName := fmt.Sprintf("%s/%s", storeTemplate.Source, storeTemplate.CliLanguage)
+				if template == storeTemplate.CliLanguage || template == sourceName {
+					if overwrite && pullDebug {
+						faasCmd.SetArgs([]string{"template", "pull", storeTemplate.Repository, "--overwrite", "--debug"})
+					} else if overwrite && !pullDebug {
+						faasCmd.SetArgs([]string{"template", "pull", storeTemplate.Repository, "--overwrite"})
+					} else if !overwrite && pullDebug {
```

Comment:
Is this style of the CLI calling itself something we want to do or should the method inside of the "template pull" be extracted and exposed? The only other place we do something like this is in tests.

Reply:
Alex was more for moving the method out and mentioned briefly that I will do it that way and leave for the community to give me feedback on the approach, I also think the method should be extracted and applied to this command, I can totally do that

Reply:
Thanks for the suggestion @LucasRoesler , I amended fix which references that, now I don't use the command calling itself, but instead the pull function